File: InPlan.Web/Modules/Tenencia/Empresas/EmpresasForm.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace InPlan.Tenencia.Forms;

[FormScript("Tenencia.Empresas")]
[BasedOnRow(typeof(EmpresasRow), CheckNames = true)]
public class EmpresasForm
{
    [DisplayName("Nombre")]
    public string DescripcionCorta { get; set; }
    public string Descripcion { get; set; }
    [TextAreaEditor(Rows = 5)]
    public string Observaciones { get; set; }
    [HideOnInsert]
    public bool Baja { get; set; }
}

File: InPlan.Web/Modules/ServerTypes/Tenencia/EmpresasForm.ts
﻿import { StringEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface EmpresasForm {
    DescripcionCorta: StringEditor;
    Descripcion: StringEditor;
    Observaciones: StringEditor;
    Baja: BooleanEditor;
}

export class EmpresasForm extends PrefixedContext {
    static formKey = 'Tenencia.Empresas';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmpresasForm.init)  {
            EmpresasForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;

            initFormType(EmpresasForm, [
                'DescripcionCorta', w0,
                'Descripcion', w0,
                'Observaciones', w0,
                'Baja', w1
            ]);
        }
    }
}